dockstore-common/src/main/scala/io/dockstore/common/WdlBridge.scala

```diff
@@ -52,16 +52,37 @@ class WdlBridge {
     secondaryWdlFiles = secondaryFiles
   }
 
+  /**
+   *
+   * Validates that the workflow or tool given by filePath has a workflow declaration
+   * @param filePath absolute path to file
+   */
+  @throws(classOf[WdlParser.SyntaxError])
+  def validateWorkflowDeclaration(filePath: String, sourceFilePath: String) = {
+    val bundle = getBundle(filePath, sourceFilePath)
+
+    if (!bundle.primaryCallable.isDefined) {
+      throw new WdlParser.SyntaxError("Workflow is missing a workflow declaration.")
+    }
+  }
+
   /**
     * Validates the workflow given by filePath
     * @param filePath absolute path to file
     */
   @throws(classOf[WdlParser.SyntaxError])
   def validateWorkflow(filePath: String, sourceFilePath: String) = {
-    val bundle = getBundle(filePath, sourceFilePath)
+    val executableCallable = convertFilePathToExecutableCallable(filePath, sourceFilePath)
+    val numberOfTaskCalls = executableCallable.taskCallNodes.seq.size
 
-    if (!bundle.primaryCallable.isDefined) {
-      throw new WdlParser.SyntaxError("Workflow is missing a workflow declaration.")
+    validateWorkflowDeclaration(filePath, sourceFilePath)
+
+    if (numberOfTaskCalls <= 1) {
+      val validationMessage = new StringBuilder("A WDL workflow must call more than one task.")
```

Comment:
This is probably the fault of our definition, but while a WDL tool should be enforced to only have one task. 
I don't know if we want to enforce that a WDL workflow should have more than one task. 
I realize that this means that one-task workflows will basically count as "both", but I think that will be better because it won't confuse users who want to register and launch-with one-step workflows.

Reply:
Note that we can/should enforce this for CWL because of the much clearer definition

Reply:
Should I remove the error message for WDL workflows then?

Additionally, how should WDL workflows and tools with zero tasks be handled? For example, if we enforce that a WDL tool should only have one task, would a WDL tool with zero tasks throw an error?

Reply:
Yeah, I think WDL workflows should be fine with 1 or many tasks. 

I would have thought that a WDL workflow or tool with zero tasks would be an error either way. But it might be worth looking at a few examples to double check. Or writing a few and having https://cromwell.readthedocs.io/en/develop/WOMtool/ validate them

Reply:
I reverted the WDL workflow validation back to what it was originally so WDL tools with 1 task call can be registered as a workflow.